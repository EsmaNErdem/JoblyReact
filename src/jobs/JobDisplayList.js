import JobCard from "./JobCard";

const JobDisplayList = () => {

}

export default JobDisplayList