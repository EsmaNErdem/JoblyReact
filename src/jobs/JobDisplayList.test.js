import { render } from "@testing-library/react";
import JobDisplayList from './JobDisplayList';
import { UserProvider } from "../testUtils";

const jobs = [
    { id: 1, title: "J1", equity: "0.1", salary: 1, companyName: "C1" },
    { id: 2, title: "J2", equity: "0.2", salary: 2, companyName: "C2" },
    { id: 3, title: "J3", equity: null, salary: 300000, companyName: "C1" },
  ];

test("renders without crashing", () => {
    render(
        <UserProvider>
                <JobDisplayList jobs={jobs} />
        </UserProvider>
    );
});

test("it renders and matches with snaphot", () => {
    const { asFragment } = render(
        <UserProvider>
                <JobDisplayList jobs={jobs} />
        </UserProvider>
    );
    expect(asFragment()).toMatchSnapshot();
});

test("Display job data", () => {
    const { getByText } = render(
        <UserProvider>
                <JobDisplayList jobs={jobs} />
        </UserProvider>
        );
    
    expect(getByText("J1")).toBeInTheDocument()
    expect(getByText("C2")).toBeInTheDocument()
    expect(getByText("Salary: $300,000")).toBeInTheDocument()
});
