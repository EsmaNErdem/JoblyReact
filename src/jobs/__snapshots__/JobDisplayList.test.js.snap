// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders and matches with snaphot 1`] = `
<DocumentFragment>
  <div
    class="JobDisplayList"
  >
    <div
      class="JobCard"
    >
      <h2
        data-testid="job-title"
      >
        J1
      </h2>
      <h3
        data-testid="company-name"
      >
        C1
      </h3>
      <p>
        Salary: $1
      </p>
      <p>
        Equity: 0.1
      </p>
      <button>
        APPLY
      </button>
    </div>
    <div
      class="JobCard"
    >
      <h2
        data-testid="job-title"
      >
        J2
      </h2>
      <h3
        data-testid="company-name"
      >
        C2
      </h3>
      <p>
        Salary: $2
      </p>
      <p>
        Equity: 0.2
      </p>
      <button>
        APPLY
      </button>
    </div>
    <div
      class="JobCard"
    >
      <h2
        data-testid="job-title"
      >
        J3
      </h2>
      <h3
        data-testid="company-name"
      >
        C1
      </h3>
      <p>
        Salary: $300,000
      </p>
      <button>
        APPLY
      </button>
    </div>
  </div>
</DocumentFragment>
`;
